README.md

```diff
@@ -12,58 +10,76 @@ The `MATLAB.gitlab-ci.yml` [template](https://gitlab.com/gitlab-org/gitlab/-/blo
+    entrypoint: [""]
+  variables:
+    MLM_LICENSE_FILE: 27000@MyLicenseServer  # Replace the value with the port number and DNS address for your network license manager
+
+```
 
 ## Examples
-Each of these examples shows how to specify the contents of your `.gitlab-ci.yml` file by using one of the jobs in the template.
+Each of these examples shows how to specify the contents of a job in your `.gitlab-ci.yml`.
 
 ### Run a Script
 Run the commands in a file named `myscript.m` in the root of your repository.
 
 ```yaml
 command:
+   extends: .matlab_defaults
    script: matlab -batch "myscript"
 ```
 ### Run Tests
 Run the tests in your [MATLAB project](https://www.mathworks.com/help/matlab/projects.html) and fail the build if any of the tests fail.
 
 ```yaml
 test:
+  extends: .matlab_defaults
   script: matlab -batch "results = runtests('IncludeSubfolders',true), assertSuccess(results);"
 ```
 ### Run Tests and Generate Artifacts
 Run the tests in your MATLAB project, and generate a JUnit test results report and a Cobertura code coverage report. Fail the build if any of the tests fail.
 
 ```yaml
-tests_artifacts:
+test_artifacts:
+  extends: .matlab_defaults
   script: |
-       matlab -batch "
-       import matlab.unittest.TestRunner
-       import matlab.unittest.Verbosity
-       import matlab.unittest.plugins.CodeCoveragePlugin
-       import matlab.unittest.plugins.XMLPlugin
-       import matlab.unittest.plugins.codecoverage.CoberturaFormat
-   
-       suite = testsuite(pwd,'IncludeSubfolders',true);
-   
-       [~,~] = mkdir('artifacts');
-   
-       runner = TestRunner.withTextOutput('OutputDetail',Verbosity.Detailed);
-       runner.addPlugin(XMLPlugin.producingJUnitFormat('artifacts/results.xml'))
-       runner.addPlugin(CodeCoveragePlugin.forFolder(pwd,'IncludingSubfolders',true, ...
-           'Producing',CoberturaFormat('artifacts/cobertura.xml')))
-   
-       results = runner.run(suite)
-       assertSuccess(results);"
-        
-  artifacts:
+    cat <<- 'BLOCK' > runAllTests.m
+          import matlab.unittest.TestRunner
+          import matlab.unittest.Verbosity
+          import matlab.unittest.plugins.CodeCoveragePlugin
+          import matlab.unittest.plugins.XMLPlugin
+          import matlab.unittest.plugins.codecoverage.CoberturaFormat
+          suite = testsuite(pwd,'IncludeSubfolders',true);
+          [~,~] = mkdir('artifacts')
+          runner = TestRunner.withTextOutput('OutputDetail',Verbosity.Detailed);
+          runner.addPlugin(XMLPlugin.producingJUnitFormat('artifacts/results.xml'))
+          % Replace `pwd` with the location of the folder containing source code
+          runner.addPlugin(CodeCoveragePlugin.forFolder(pwd,'IncludingSubfolders',true, ...
+          'Producing',CoberturaFormat('artifacts/cobertura.xml')))
+          results = runner.run(suite)
+          assertSuccess(results);
+    BLOCK
```

Comment:
Why have we lost all of the empty lines in here? It's very hard to read without them.

Reply:
I just copied this code snippet from the template :(

Reply:
@nbhoski Why have all the empty lines been removed from this block in the template?

Reply:
I also don't understand why we are having an alignment mismatch:

    paths:
    - "./artifacts"
      - "./artifacts"

Reply:
Also, couldn't we use strings instead of character vectors here?

Reply:
I think we should fix the spacing here and in the template. It is very hard to read.

Reply:
@nbhoski: Could you please provide a more readable version?

Reply:
Hi, @nbhoski. Any comments? I hope we can merge this pull request shortly.

Reply:
> @nbhoski Why have all the empty lines been removed from this block in the template?

@mcafaro  I am not sure but think when i moved to the heredoc approch i reformatted this. may be because i got feedback. I understand tha with multi line ofers better redability , i would like to remain consistent with our template for now we can add those lines back during next review.